app/controllers/application_controller.rb

```diff
@@ -23,6 +23,7 @@ class ApplicationController < ActionController::API
   end
 
   before_action AuthenticateApplication
+  before_action :validate_schema
```

Comment:
One way to do it wihout doing the class self on the filter and adding a method the controller:

```
before_action ValidateSchema
```

The good thing is that Rails pass the controller to you.
Then on ValidateSchema you can move the logic there:

```ruby
class ValidateSchema
  def self.before(controller)
    return unless controller.request.post?
     schema = "request.#{controller.request.params['controller']}"
     validate(controller.request.params, schema)
  rescue JSON::Schema::ValidationError, JSON::Schema::SchemaError, SchemaNotFoundError => e
     controller.render '...'
  end

  # ...
end
```

What do you think?

Reply:
yup good idea! will update